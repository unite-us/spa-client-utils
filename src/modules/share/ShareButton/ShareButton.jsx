import React from 'react';
import { PropTypes } from 'prop-types';
import { Button, Icon } from '@unite-us/ui';
import classNames from 'classnames';

const ShareButton = ({ className, ...props }) => (
  <Button
    className={classNames('share-button', className)}
    {...props}
  />
);

ShareButton.propTypes = {
  /** class name */
  className: PropTypes.string,
  /** icon to the right of the label, defaults to IconShare */
  iconRight: PropTypes.node,
  /** Required button id */
  id: PropTypes.string.isRequired,
  /** button label, defaults to "Share" */
  label: PropTypes.string,
};

ShareButton.defaultProps = {
  className: '',
  iconRight: <Icon icon="IconShare" color="#2C405A" />,
  label: 'Share',
};

export default ShareButton;
