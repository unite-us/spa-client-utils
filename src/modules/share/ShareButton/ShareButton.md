```jsx
<ShareButton
  className="this-share-button"
  id="this-share-button-id"
  onClick={() => alert('Clicked Share Button!')}
/>
```
