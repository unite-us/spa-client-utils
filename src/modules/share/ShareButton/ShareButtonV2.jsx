import React from 'react';
import { PropTypes } from 'prop-types';
import { Button, Icon } from '@unite-us/ui';
import classNames from 'classnames';

const ShareButton = ({ className, ...props }) => (
  <Button
    className={classNames('share-button', className)}
    {...props}
  />
);

ShareButton.propTypes = {
  /** class name */
  className: PropTypes.string,
  /** icon to the left of the label, defaults to IconShare */
  iconLeft: PropTypes.node,
  /** Required button id */
  id: PropTypes.string.isRequired,
  /** button label, defaults to "Share" */
  label: PropTypes.string,
};

ShareButton.defaultProps = {
  className: '',
  iconLeft: <Icon icon="IconMailForward" color="#4571BA" />,
  label: 'Share',
};

export default ShareButton;
