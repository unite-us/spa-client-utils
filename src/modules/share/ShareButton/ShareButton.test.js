import React from 'react';
import renderer from 'react-test-renderer';
import ShareButton from './ShareButton';

describe('ShareButton', () => {
  it('renders default button with Share label', () => {
    expect(renderer.create(
      <ShareButton id="share-button-id" />,
    )).toMatchSnapshot();
  });

  it('renders share button with custom className, iconRight and label', () => {
    expect(renderer.create(
      <ShareButton
        className="this-button-class"
        iconRight={<mock-icon />}
        id="share-button-id"
        label="Custom Share Label"
      />,
    )).toMatchSnapshot();
  });
});
