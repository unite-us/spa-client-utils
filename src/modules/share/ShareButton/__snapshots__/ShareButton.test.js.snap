// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareButton renders default button with Share label 1`] = `
<button
  aria-label="Share"
  className="ui-button ui-button--icon share-button"
  data-test-element={null}
  disabled={false}
  id="share-button-id"
  onClick={[Function]}
  style={Object {}}
  type="button"
>
  Share
  <span
    className="ui-button__icon-right"
  >
    <svg
      className="ui-icon"
      height="16px"
      id=""
      style={
        Object {
          "display": "inline-block",
          "fill": "#2C405A",
          "verticalAlign": "middle",
        }
      }
      viewBox="0 0 16 16"
      width="16px"
    >
      <path
        d="M16,8 C16,12.4193548 12.4193548,16 8,16 C3.58064516,16 0,12.4193548 0,8 C0,3.58064516 3.58064516,0 8,0 C12.4193548,0 16,3.58064516 16,8 Z M7.0740732,5.58609475 L7.59625144,5.01305024 L7.59625144,9.55681012 C7.59625144,9.8015691 7.77705929,10 8.00008077,10 C8.22310224,10 8.40391009,9.8015691 8.40391009,9.55681012 L8.40391009,5.01307978 L8.92603449,5.58609475 C9.0048889,5.67263496 9.10821536,5.71591983 9.21156875,5.71591983 C9.31492213,5.71591983 9.41827552,5.6726645 9.49710301,5.58609475 C9.65481182,5.41301433 9.65481182,5.13241604 9.49710301,4.95933562 L8.28569579,3.62982509 C8.27621926,3.61945444 8.26628506,3.60964518 8.25597395,3.60036773 C8.25150491,3.59634948 8.24679356,3.59286305 8.24224375,3.58911071 C8.23626708,3.58417653 8.23042501,3.57915371 8.22420604,3.57457408 C8.2186332,3.57046719 8.21284498,3.56692167 8.2071106,3.56316933 C8.20153776,3.55947608 8.1960726,3.5556351 8.1903113,3.55226686 C8.18446924,3.54883952 8.17849256,3.54594402 8.17251589,3.54287123 C8.16651229,3.53973936 8.16064331,3.5364893 8.15447818,3.53368243 C8.14871688,3.53105284 8.14284789,3.52895507 8.13700583,3.52665048 C8.13043687,3.52405044 8.12394868,3.52127311 8.11719127,3.51905716 C8.11137613,3.51713668 8.10548022,3.51571847 8.09963816,3.51409344 C8.09274614,3.51217295 8.08590796,3.51004564 8.07888133,3.50853879 C8.07215084,3.50709104 8.06533959,3.5062342 8.05855525,3.505141 C8.05230936,3.50416598 8.04617115,3.50286596 8.03984449,3.5021864 C8.02740655,3.50085683 8.01494169,3.50020682 8.0024499,3.50011818 C8.00164224,3.50011818 8.0008615,3.5 8.00005384,3.5 C7.99927311,3.5 7.99846545,3.50011818 7.99768471,3.50011818 C7.98519292,3.50017728 7.97270114,3.50085683 7.96023627,3.5021864 C7.95404422,3.50286596 7.94804063,3.50413644 7.94192934,3.50508191 C7.93498348,3.50617511 7.92803761,3.50706149 7.92117252,3.50856834 C7.91433434,3.51004564 7.90771154,3.51211386 7.90100797,3.51397525 C7.89492361,3.51565938 7.88883925,3.51710713 7.88280873,3.51908671 C7.87632054,3.52124357 7.86999388,3.52393225 7.8636403,3.52644366 C7.85758286,3.52883689 7.85147157,3.53102329 7.8454949,3.53374152 C7.83957207,3.53643021 7.83391846,3.53959163 7.82815716,3.54257577 C7.82196511,3.54576674 7.81571922,3.54880998 7.80963485,3.5523555 C7.80414278,3.55557601 7.79897376,3.55923971 7.79361629,3.56275569 C7.7876127,3.56665576 7.78158218,3.57037855 7.77574011,3.57466272 C7.76989805,3.57894689 7.76443289,3.58373334 7.7588062,3.58831297 C7.75385256,3.59236077 7.74881816,3.59611311 7.74402605,3.60045637 C7.73419954,3.60929062 7.72480377,3.6186567 7.71578492,3.62843643 C7.71535417,3.62890916 7.71489649,3.62929326 7.71443882,3.629766 L6.50289699,4.95933562 C6.34518818,5.13241604 6.34518818,5.41301433 6.50289699,5.58609475 C6.66068657,5.75917517 6.9163913,5.75911607 7.0740732,5.58609475 Z M10.6086957,6.5 L9.56521739,6.5 C9.34911304,6.5 9.17391304,6.69188571 9.17391304,6.92857143 C9.17391304,7.16525714 9.34911304,7.35714286 9.56521739,7.35714286 L10.2173913,7.35714286 L10.2173913,11.6428571 L5.7826087,11.6428571 L5.7826087,7.35714286 L6.43478261,7.35714286 C6.65088696,7.35714286 6.82608696,7.16525714 6.82608696,6.92857143 C6.82608696,6.69188571 6.65088696,6.5 6.43478261,6.5 L5.39130435,6.5 C5.1752,6.5 5,6.69188571 5,6.92857143 L5,12.0714286 C5,12.3081143 5.1752,12.5 5.39130435,12.5 L10.6086957,12.5 C10.8248,12.5 11,12.3081143 11,12.0714286 L11,6.92857143 C11,6.69188571 10.8248,6.5 10.6086957,6.5 Z"
      />
    </svg>
  </span>
</button>
`;

exports[`ShareButton renders share button with custom className, iconRight and label 1`] = `
<button
  aria-label="Custom Share Label"
  className="ui-button ui-button--icon share-button this-button-class"
  data-test-element={null}
  disabled={false}
  id="share-button-id"
  onClick={[Function]}
  style={Object {}}
  type="button"
>
  Custom Share Label
  <span
    className="ui-button__icon-right"
  >
    <mock-icon />
  </span>
</button>
`;
